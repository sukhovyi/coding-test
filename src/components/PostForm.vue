<template>
  <div>
    <div>
      <input type="text"/>
    </div>
    <div>
      <textarea/>
    </div>
  </div>
</template>

<script>
import UiButton from '../ui-kit/ui-button/UiButton';

export default {
  name: 'PostForm',
  components: { UiButton },
};
</script>
