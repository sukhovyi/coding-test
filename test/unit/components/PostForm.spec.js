import { shallowMount } from '@vue/test-utils';
import PostForm from '../../../src/components/PostForm';

describe('components/PostForm', () => {
  const getComponent = post => shallowMount(PostForm, {
    propsData: {
      post,
    },
  });

  it('should be rendered correctly for new post', () => {
    expect(getComponent({}).element).toMatchSnapshot();
  });

  it('should be rendered correctly for edit post', () => {
    expect(getComponent({
      id: 123,
      title: 'test title',
      body: 'test description',
    }).element).toMatchSnapshot();
  });

  it('should not save invalid post', () => {
    const wrapper = getComponent({});
    wrapper.vm.save();
    expect(wrapper.vm.error).toEqual({
      body: ['Body is a required field'],
      title: ['Title is a required field'],
    });
  });

  it('should save valid post', () => {
    const wrapper = getComponent({
      title: 'test title',
      body: 'test description',
    });

    wrapper.vm.save();
    expect(wrapper.emitted('save')).toEqual([[{
      title: 'test title',
      body: 'test description',
    }]]);
  });
});
