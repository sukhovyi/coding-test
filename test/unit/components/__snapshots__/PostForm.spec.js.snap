// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PostForm should be rendered correctly for edit post 1`] = `
<div>
  <div>
    <ui-input-stub
      debouncetime="0"
      errors=""
      hints=""
      label="Post Title"
      placeholder="Title"
      type="text"
      value="test title"
    />
  </div>
   
  <div>
    <ui-textarea-stub
      errors=""
      hints=""
      label="Post description"
      placeholder="Description"
      value="test description"
    />
  </div>
   
  <div>
    <ui-button-stub
      type="primary"
    >
      
      Save Changes
    
    </ui-button-stub>
     
    <ui-button-stub
      type="secondary"
    >
      
      Cancel
    
    </ui-button-stub>
  </div>
</div>
`;

exports[`components/PostForm should be rendered correctly for new post 1`] = `
<div>
  <div>
    <ui-input-stub
      debouncetime="0"
      errors=""
      hints=""
      label="Post Title"
      placeholder="Title"
      type="text"
    />
  </div>
   
  <div>
    <ui-textarea-stub
      errors=""
      hints=""
      label="Post description"
      placeholder="Description"
    />
  </div>
   
  <div>
    <ui-button-stub
      type="primary"
    >
      
      Create Post
    
    </ui-button-stub>
     
    <ui-button-stub
      type="secondary"
    >
      
      Cancel
    
    </ui-button-stub>
  </div>
</div>
`;
